### PR TITLE
show a message when nothing exists to migrate

### PR DESCRIPTION
**Pull request by Pentarctagon**

### CODE REVIEW

File: src/gui/dialogs/migrate_version_selection.cpp [hunk 1]
```diff
@@ -43,6 +44,8 @@ void migrate_version_selection::execute()
 	migrate_version_selection mig = migrate_version_selection();
 	if(mig.versions_.size() > 0) {
 		mig.show();
+	} else {
+		gui2::show_message(_("No Versions"), _("There are no other previous versions from which data can be migrated from"), gui2::dialogs::message::button_style::auto_close);
```
Comment (stevecotton) [inline src/gui/dialogs/migrate_version_selection.cpp:48]: One of those "from"s should be removed

Comment (Pentarctagon) [reply src/gui/dialogs/migrate_version_selection.cpp]: Updated.

File: src/gui/dialogs/migrate_version_selection.cpp [hunk 2]
```diff
@@ -43,6 +44,8 @@ void migrate_version_selection::execute()
 	migrate_version_selection mig = migrate_version_selection();
 	if(mig.versions_.size() > 0) {
 		mig.show();
+	} else {
+		gui2::show_message(_("No Versions"), _("There are no other previous versions which data can be migrated from"), gui2::dialogs::message::button_style::auto_close);
```
Comment (stevecotton) [inline src/gui/dialogs/migrate_version_selection.cpp:48]: Having both "other" and "previous" seems redundant.

**Comment by stevecotton** [comment]
This text assumes that the user already understands what the button is meant to do, but seeing this error likely means that they pushed the button wondering what it did, as I'd expect them to know whether they'd played an old version on a PC (recently enough that they haven't customised the new install.

Instead of making this a new pop-up, I think it would be better to show the migration dialog, thus getting the explanation that's already in that dialog, and put the error message in place of the list (IIRC some of the list widgets have already support for a string to show when they're empty), along with greying out the "OK" button.

**Comment by Pentarctagon** [comment]
I think if they clicked a button called Migrate which has the associated text "Re-open the version migrator dialog" and the message added here says that saying that there are no previous versions to migrate data from, then it's very clear what's being talked about.

**Comment by Pentarctagon** [comment]
That sounds like something that would be better shown as a tooltip rather than opening an empty dialog that they can't do anything with.

**Comment by Pentarctagon** [comment]
Also, I'm not sure what different information the dialog is giving that the new popup isn't. The popup says:
> There are no other previous versions which data can be migrated from

The dialog says:
> Choose which version of Wesnoth to import your data from

Aside from "migrate" vs "import", they give essentially the same information to someone who has no idea what it does.

**Comment by stevecotton** [comment]
I think just using a different word to "migrate" helps. Maybe use this wording instead:

"No Other Version Found", "This would import settings from a previous version of Wesnoth, but no other version was found on this device."

Either way, I'll stop blocking this. Please merge, either with or without the suggested change.

**Comment by Pentarctagon** [comment]
Updated to that wording.

**Pull request merged**